### DISCUSSION TITLE
FriendlyFire="False" doesnt work the expected way

### DISCUSSION THREAD

**Discussion by PanmanS** [FakeFishGames/Barotrauma#15431] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

So basically, we tried to create  homing bullets using defensebot attack. Basically shoot bullet - it spawns creature - creature shoots at enemy. But for some reason with friendlyfire="false" bullets stop hitting bandits, despite using same ammo with same settings working just fine if it by player.

### Reproduction steps

1. download [Hellsmg.zip](https://github.com/user-attachments/files/18222338/Hellsmg.zip)
2. launch any sub and get yourself deadeye carbine + smg ammo, insert ammo, then click homing button
3. spawn security officer in team 0 
4. try to shoot him and observe how whole barrage goes through seemingly enemy guy (ignore 1 bullet in end that hits him lol)

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.7.7.0 (Winter Update)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report!

I gave the mod a test, but unfortunately couldn't figure out how to get it working. Clicking the "homing" button plays a beeping sound, but nothing else seems to happen.

**Reply by PanmanS** [reply]
Clicking himing button turns smg into homing mod, holding fire button starts firing sequence of self-homing amm0

**Reply by PanmanS** [reply]
We discovered that creature part of bullet takes team from SMG itself, so team none/0. Fix for issue to forcibly apply team id of crew and that made bullet work as intended if used in campaign by player. Consequences of such solution is that if you not in player crew a.k.a team 1, you kill yourself upon shooting since creature bullet considers you to be enemy.

**Reply by PanmanS** [reply]
Sorry for long reply time, Im very unused to checking email for notifs instead of something like discord or phone pushup.
